text/0103-introduce-core-index-commitment.md

```diff
@@ -0,0 +1,192 @@
+
+Once this RFC is implemented the validity of a parachain block depends on the core it is being executed on.
+
+## Stakeholders
+
+- Polkadot core developers.
+- Cumulus node developers.
+- Tooling, block explorer developers.
+
+This approach and alternatives have been considered and discussed in [this issue](https://github.com/polkadot-fellows/RFCs/issues/92).
+
+## Explanation
+
+The approach proposed below was chosen primarly because it minimizes the number of breaking changes, the complexity and takes far less implementation and testing time. The proposal is to free up space and introduce a new core index field in the `CandidateDescriptor` primitive and use the UMP queue as output for `CoreIndex` commitment. 
+
+### Reclaiming unused space in the descriptor
+The `CandidateDescriptor` currently includes `collator` and `signature` fields. The collator includes a signature on the following descriptor fields: parachain id, relay parent, validation data hash, validation code hash and the PoV hash.
+
+However, in practice, having a collator signature in the receipt on the relay chain does not provide any benefits as there is no mechanism to punish or reward collators that have provided bad parachain blocks.
+
+This proposal aims to remove the two fields and all the logic that checks the collator signatures. We reclaim the unused space as `reserved` fields and fill it with zeroes, so there is no change in the layout and lenght of the receipt. The new primitive binary compatible with the old one.
+
+
+### Backwards compatibility
+
+There are two flavors of candidate receipts which are used in network protocols, runtime and node implementation:
+- `CommittedCandidateReceipt` which includes the `CanidateDescriptor` and the `CandidateCommitments` 
+- `CandidateReceipt` which includes the `CanidateDescriptor` and just a hash of the commitments
+
+We want to support both the old and new versions in the runtime and node . The implementation must be able to detect the version of a given candidate receipt.
+
+This is easy to do in both cases:
+- the reserved fields are zeroed
+- the UMP queue contains the core index commitment that matches the core index in the descriptor.
+
+
+### Polkadot Primitive changes
+
+#### New [CandidateDescriptor](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/primitives/src/v7/mod.rs#L482)
+
+- reclaim 32 bytes from `collator: CollatorId` and 64 bytes from `signature: CollatorSignature` as `reserved` 
+- use 4 bytes for a new `core_index: CoreIndex` field.
+- the unused reclaimed space will be filled with zeroes
+
+Thew new primitive will look like this:
+```
+pub struct CandidateDescriptor<H = Hash> {
+	/// The ID of the para this is a candidate for.
+	pub para_id: Id,
+	/// The hash of the relay-chain block this is executed in the context of.
+	pub relay_parent: H,
+	/// The core index where the candidate is backed.
+	pub core_index: CoreIndex,
```

Comment:
This should be private. We must now use accessor functions and should enforce their usage. Those accessor functions will read the version part and either return the data here or `None` in case the version is not `0`.

Reply:
I will add a section to describe the new interface.